Overset mesh simulation using TIOGA TPL

sayerhs opened the pull request:
Introduces ability to perform generalized overset mesh simulations using TIOGA TPL library. 

- Refactor `OversetManager` class to allow runtime selection of classic STK based overset hole cutting (`OversetManagerSTK`) and TIOGA based overset assembly (`OversetManagerTIOGA`). 

- Introduces a CMake option `ENABLE_TIOGA` (default: `OFF`) that is necessary to activate TIOGA during compile time. Uses a compiler definition `NALU_USES_TIOGA` to conditionally compile all TIOGA related sources; when undefined adds error checking messages if TIOGA is selected during runtime. 

- Introduces a new `overset_connectivity_type` flag to select TIOGA during runtime. If not present in input file, defaults to STK overset manager retaining backwards compatibility. 

- Expanded theory manual to discuss TIOGA overset and introduced a regression test for TIOGA based overset assembly. New mesh introduced in NaluCFD/NaluMesh repo, the submodule version will be updated with the merge of this pull request. 

- Added a new spack compilation package in NaluCFD/NaluSpack to allow building TIOGA with spack and adding TIOGA as a build variant with spack.

spdomin commented:
I had a comment above about a possible base class method. Also, some offline discussion of the polymorphic design attribute.

aprokop commented on docs/source/theory/overset.rst:
```diff
@@ -98,9 +122,137 @@ while background is hex) is provided in Figure :numref:`overset-sphere`.
+strategies.
+
+TIOGA uses a different nomenclature for overset mesh assembly. A brief
+description is provided here to familiarize users with the differnces in
```
differnces -> differences. May be a good idea to run spellcheck.

sayerhs commented:
@spdomin I have removed the redundant methods from the subclasses in the latest commit 510b155

@aprokop thanks for pointing out the typo, I've fixed it.
